parse/parser.go

```diff
@@ -200,16 +202,44 @@ func makePrefixExpression(buf TokenBuffer) (ast.Expression, []error) {
 	return exp, errs
 }
 
-// TODO: implement me w/ test cases :-)
-// MakeInfixExpression retrieves infix parse function from map
-// then parse expression with that function if exist.
 func makeInfixExpression(buf TokenBuffer, exp ast.Expression, pre precedence) (ast.Expression, []error) {
-	return nil, nil
+	errs := make([]error, 0)
+	expression := exp
+	curTok := buf.Peek(CURRENT)
+
+	for pre < nextPrecedence(buf) {
+		fn := infixParseFnMap[buf.Peek(NEXT).Type]
+
+		if fn == nil {
+			errs = append(errs, parseError{curTok.Type, "infix parse function not defined"})
```

Comment:
We don't need `curTok` except for this error message. so how about remove declaration curTok in line 208? And just `buf.Peek(CURRENT)` here